src/utils/polygon.cpp

```diff
@@ -597,6 +597,81 @@ void Polygons::removeSmallAreas(const double min_area_size, const bool remove_ho
     paths.resize(new_end - paths.begin());
 }
 
+void Polygons::removeSmallCircumference(const coord_t min_circumference_size, const bool remove_holes)
+{
+    Polygons new_polygon;
+
+    bool outline_is_removed = false;
+    for (ConstPolygonRef poly : paths)
+    {
+        bool is_outline = poly.area() >= 0;
+        auto circumference = poly.polygonLength();
+
+        if (is_outline)
+        {
+            if (circumference >= min_circumference_size)
+            {
+                new_polygon.add(poly);
+                outline_is_removed = false;
+            }
+            else
+            {
+                outline_is_removed = true;
+            }
+        }
+        else if (outline_is_removed)
+        {
+            // containing parent outline is removed; hole should be removed as well
+        }
+        else if (!remove_holes || circumference >= min_circumference_size)
+        {
+            // keep hole-polygon if we do not remove holes, or if its
+            // circumference is bigger then the minimum circumference size
+            new_polygon.add(poly);
+        }
+    }
+
+    *this = new_polygon;
+}
+
+void Polygons::removeSmallAreaCircumference(const double min_area_size, const coord_t min_circumference_size, const bool remove_holes)
```

Comment:
I see a lot of code duplication. Please mkae this function a specialization of `removeSmallAreaCircumference`

Reply:
Ofcourse I mean the function above ;-)

Reply:
https://github.com/Ultimaker/CuraEngine/pull/1803/commits/f0f49c17ac53216d7dc77fcbc63d688c5aca5a13